massa-pos-exports/src/cycle_info.rs

```diff
@@ -0,0 +1,322 @@
+}
+
+impl Deserializer<PreHashMap<Address, ProductionStats>> for ProductionStatsDeserializer {
+    fn deserialize<'a, E: ParseError<&'a [u8]> + ContextError<&'a [u8]>>(
+        &self,
+        buffer: &'a [u8],
+    ) -> IResult<&'a [u8], PreHashMap<Address, ProductionStats>, E> {
+        context(
+            "Failed ProductionStats deserialization",
+            length_count(
+                context("Failed length deserialization", |input| {
+                    self.u64_deserializer.deserialize(input)
+                }),
+                tuple((
+                    context("Failed address deserialization", |input| {
+                        self.address_deserializer.deserialize(input)
+                    }),
+                    context("Failed block_success_count deserialization", |input| {
+                        self.u64_deserializer.deserialize(input)
+                    }),
+                    context("Failed block_failure_count deserialization", |input| {
+                        self.u64_deserializer.deserialize(input)
+                    }),
+                )),
+            ),
+        )
+        .map(|elements| {
+            elements
+                .into_iter()
+                .map(|(addr, block_success_count, block_failure_count)| {
+                    (
+                        addr,
+                        ProductionStats {
+                            block_success_count,
+                            block_failure_count,
+                        },
+                    )
+                })
+                .collect()
+        })
+        .parse(buffer)
+    }
+}
+
+/// Deserializer for rolls
+pub struct RollsDeserializer {
+    address_deserializer: AddressDeserializer,
+    u64_deserializer: U64VarIntDeserializer,
+}
+
+impl RollsDeserializer {
+    /// Creates a new rolls deserializer
+    pub fn new(max_rolls_length: u64) -> RollsDeserializer {
+        RollsDeserializer {
+            address_deserializer: AddressDeserializer::new(),
+            u64_deserializer: U64VarIntDeserializer::new(
+                Included(u64::MIN),
+                Included(max_rolls_length),
+            ),
+        }
+    }
+}
+
+impl Deserializer<Vec<(Address, u64)>> for RollsDeserializer {
+    fn deserialize<'a, E: ParseError<&'a [u8]> + ContextError<&'a [u8]>>(
+        &self,
+        buffer: &'a [u8],
+    ) -> IResult<&'a [u8], Vec<(Address, u64)>, E> {
+        context(
+            "Failed rolls deserialization",
+            length_count(
+                context("Failed length deserialization", |input| {
+                    self.u64_deserializer.deserialize(input)
+                }),
+                tuple((
+                    context("Failed address deserialization", |input| {
+                        self.address_deserializer.deserialize(input)
+                    }),
+                    context("Failed number deserialization", |input| {
+                        self.u64_deserializer.deserialize(input)
```

Comment:
The deserializer you are using here have a limit on the rolls length so it will limit this field also